### PR TITLE
Update csproj files to the latest version of the package references

### LINKED ISSUE #5957: create a script to update nuget packages to latest.

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Update csproj files to the lates version of the package references

Fixes  #5957 

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [NA] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [NA] Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

**Comment by bergmeister** [comment]
Is this PR also to close  #5957 ?

File: tools/releaseTools.psm1 [hunk 1]
```diff
@@ -189,4 +189,62 @@ function Get-ChangeLog
...
+        foreach($package in $packages)
+        {
+            # Get the name of the package
+            $name = $package.Include
+
+            # don't pull 'Microsoft.Management.Infrastructure' from nuget
+            if($name -and $name -ne 'Microsoft.Management.Infrastructure')
+            {
+                # Get the current package from nuget
+                $newPackage = find-package -Name $name -Source https://nuget.org/api/v2/  -ErrorAction SilentlyContinue
```
Comment (bergmeister) [inline tools/releaseTools.psm1]: 'donet add package' apparently handles updating nuget packages as well. We should not re-invent the wheel.  https://ardalis.com/how-do-i-update-a-nuget-package-using-dotnet-cli

File: tools/releaseTools.psm1 [hunk 2]
```diff
@@ -189,4 +189,62 @@ function Get-ChangeLog
...
+#Objects which represet the csproj package ref, with the current and new version
+##############################
+function Get-NewPackage
+{
+    param(
+        [String]
+        $Path = (Join-path -Path $PSScriptRoot -ChildPath '..')
+    )
+    # Calculate the filter to find the CSProj files
+    $path = Join-path -Path $PSScriptRoot -ChildPath '..'
```
Comment (daxian-dbw) [inline tools/releaseTools.psm1]: You override the path argument passed in by the caller.

File: tools/releaseTools.psm1 [hunk 3]
```diff
@@ -189,4 +189,62 @@ function Get-ChangeLog
...
+    $filter = Join-Path -Path $path -ChildPath '*.csproj'
```
Comment (daxian-dbw) [inline tools/releaseTools.psm1]: Minor: change `$path` to `$Path` because the parameter is `$Path`

**Comment by TravisEz13** [comment]
@bergmeister  I reverted some of my changes and tried `donet add package` and it did not update the reference.

**Comment by iSazonov** [comment]
What is our policy - how frequently we should check existence of new packages ?

**Review by iSazonov — APPROVED** [review]

**Review by daxian-dbw — COMMENTED** [review]
LGTM. Left one unblocking comment

**Review by adityapatwardhan — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@iSazonov The guideline in practice is:
- When moving powershell to target a preview version of .NET Core runtime, all package references need to be updated to use the versions come from the same build of the .NET Core runtime (will be from the dotnet-core myget feed). This is to guarantee that all packages are in sync.
- When moving powershell to target an official .NET Core runtime, all package references need to be updated to use the latest stable versions from nuget.org.

**Pull request merged**